### PR TITLE
refactor: refactor tooltip components

### LINKED ISSUE #3456: Rename nzTitle based on the directive
## What problem does this feature solve?
Having nz-tooltip, nz-popconfirm and nz-popover on the same component makes some behaviors impossible because both of them use the same input for the title: `nzTitle`.

This means that if you want to create a button that has a tooltip and a popover on click, you won't be able to show a tooltip at all.
## What does the proposed API look like?
Use `nzPopoverTitle`, `nzPopConfirmTitle` and `nzTooltipTitle` as optional second input, with an implicit setter in order to map these to the effective input value and minimize the impact of this change while providing a way to solve this issue.

I'm not sure these are the only directives that should be modified, but they are the ones I spotted so far.

### PR DESCRIPTION
**Pull request by wendzhue**
close #3456 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### TODO

- [x] Use titles with prefix (like `nzTooltipTitle`) and deprecate old ones without prefix
- [x] Typing things
- [x] And new api for positions and trigger, too.
- [x] Add deprecation information.

Component usage will be removed in 9.0.0. 

API without prefix will be removed in 10.0.0.

### CODE REVIEW

File: components/popconfirm/doc/index.en-US.md [hunk 1]
```diff
@@ -26,10 +26,12 @@ import { NzPopconfirmModule } from 'ng-zorro-antd/popconfirm';
 
 | Param | Description | Type | Default value |
 | ----- | ----------- | ---- | ------------- |
+| `[nzTitle]` | Title of the confirmation box | `string \| TemplateRef<void>` | - |
```
Comment (hsuanxyz) [inline components/popconfirm/doc/index.en-US.md]:
```suggestion
| `[nzPopconfirmTitle]` | Title of the confirmation box | `string \| TemplateRef<void>` | - |
```

File: components/popconfirm/demo/basic.ts [hunk 2]
```diff
@@ -4,7 +4,15 @@ import { NzMessageService } from 'ng-zorro-antd';
 @Component({
   selector: 'nz-demo-popconfirm-basic',
   template: `
-    <a nz-popconfirm nzTitle="Are you sure delete this task?" (nzOnConfirm)="confirm()" (nzOnCancel)="cancel()"
+    <a
+      nz-popconfirm
+      nzPopconfirmTitle="Are you sure delete this task?"
+      nzPopconfirmPlacement="bottom"
+      nz-tooltip
```
Comment (hsuanxyz) [inline components/popconfirm/demo/basic.ts]: 如果是为了测试最好写测试里面，在 basic DEMO 里面不合适

File: components/tooltip/base/nz-tooltip-base.directive.ts [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+        this.updateComponentValue('nzPlacement', this.placement);
+      }
+    }
+
+    this.tooltip.updateByDirective();
+  }
+
+  // tslint:disable-next-line no-any
+  private updateComponentValue(key: string, value: any): void {
+    if (isNotNil(value)) {
```
Comment (hsuanxyz) [inline components/tooltip/base/nz-tooltip-base.directive.ts]: 这里的判断应该可以不要，再确认一下

File: components/tooltip/doc/index.en-US.md [hunk 4]
```diff
@@ -37,11 +40,15 @@ The following APIs are shared by `nz-tooltip`, `nz-popconfirm`, `nz-popover`.
...
-| `[nzPlacement]` | The position of the tooltip relative to the target | `'top' \| 'left' \| 'right' \| 'bottom' \| 'topLeft' \| 'topRight' \| 'bottomLeft' \| 'bottomRight' \| 'leftTop' \| 'leftBottom' \| 'rightTop' \| 'rightBottom'` | `'top'` |
-| `[nzTrigger]` | Tooltip trigger mode | `'click' \| 'focus' \| 'hover'` | `'hover'` |
 | `[nzVisible]` | Whether the floating tooltip card is visible or not | `boolean` | `false` |
 | `(nzVisibleChange)` | Callback executed when visibility of the tooltip card is changed | `EventEmitter<boolean>` | - |
 
+| Method | Description |
+| --- | --- |
+| `show` | Show |
+| `hide` | Hide |
+| `updatePosition` | Force to update position |
```
Comment (hsuanxyz) [inline components/tooltip/doc/index.en-US.md]:
```suggestion
| `updatePosition` | Update position |
```

File: components/tooltip/doc/index.zh-CN.md [hunk 5]
```diff
@@ -39,11 +42,15 @@ import { NzToolTipModule } from 'ng-zorro-antd/tooltip';
...
-| `[nzPlacement]` | 气泡框位置 | `'top' \| 'left' \| 'right' \| 'bottom' \| 'topLeft' \| 'topRight' \| 'bottomLeft' \| 'bottomRight' \| 'leftTop' \| 'leftBottom' \| 'rightTop' \| 'rightBottom'` | `'top'` |
-| `[nzTrigger]` | 触发行为，可选 `hover/focus/click` | `'click' \| 'focus' \| 'hover'` | `'hover'` |
 | `[nzVisible]` | 用于手动控制浮层显隐 | `boolean` | `false` |
 | `(nzVisibleChange)` | 显示隐藏的回调 | `EventEmitter<boolean>` | - |
 
+| 方法 | 说明 |
+| --- | --- |
+| `show` | 打开 |
+| `hide` | 隐藏 |
+| `updatePosition` | 强制调整位置 |
```
Comment (hsuanxyz) [inline components/tooltip/doc/index.zh-CN.md]:
```suggestion
| `updatePosition` | 调整位置 |
```

File: components/tooltip/base/nz-tooltip-base.directive.ts (hunk 3 above)
Comment (wendzhue) [reply components/tooltip/base/nz-tooltip-base.directive.ts]: 得留下，不然 popconfirm 会因为内容为空无法展开

**Pull request merged**